### PR TITLE
Extend overlay (work from AD2076 and AmonRaNet)

### PR DESCRIPTION
**Pull request by AmonRaNet**
These are combined changes from  `AD2076` fork and `AmonRaNet` fork.

This includes:

General re-design:
- amiibo icon preview
- new toggle concept (individual for emulation and activation)
- favorites support

Focus on active amiibo:
- when overlay opens first time: "View" will guide to location of active amiibo
- every time when sub-view opens folder of active amiibo will be in focus

Translation support:
- Qt like style (base strings in code)
- fast update/create of translation with: make translation
- added translation for _ru_ , _de_, _es_ (thanks to @impeeza)

For testing: 
[11.x overlays.zip](https://github.com/XorTroll/emuiibo/files/6289013/overlays.zip)
[12.x - 13.x overlays.zip](https://github.com/XorTroll/emuiibo/files/6332401/overlays.zip)

If this PR approved and merged, then PRs can be **declined** (because they are full or partial subset of this change):
[PR 80](https://github.com/XorTroll/emuiibo/pull/80)
[PR 163](https://github.com/XorTroll/emuiibo/pull/163)

_I hope `AD2076` is not against this PR (I will inform him)!_

P.S. Here `AD2076` tesla extension still part of repo, because submodule version still missing support of `HidNpadButton_`. Technically is copy from [libtesla_extensions](https://github.com/AD2076/libtesla_extensions) with fixes for `HidNpadButton_` .

2021011211231800-02CB906EA538A35643C1E1484C4B947D

### CODE REVIEW

File: overlay/emuiibo/lng_de.json [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        },
+        {
+            "context": "",
+            "source": "Emulation status",
+            "translation": "Emulationsstatus"
+        },
+        {
+            "context": "",
+            "source": "on",
+            "translation": "auf"
```
Comment (lookye) [inline overlay/emuiibo/lng_de.json:117]: Imo this should be "an" not "auf" because the emulation is "an" not "auf" :)

Comment (AmonRaNet) [reply overlay/emuiibo/lng_de.json]: Fixed

**Comment by jacoghi** [comment]
Doesn't seem to be working with AMS 0.19.2 (?)

**Comment by AmonRaNet** [comment]
> Doesn't seem to be working with AMS 0.19.2 (?), but looks great nevertheless.

Also this [12.x overlays.zip](https://github.com/XorTroll/emuiibo/files/6332401/overlays.zip) ?

**Comment by impeeza** [comment]
For me 12_X_overlays.zip works fine on Atmosphère 0.19.2 and Firmware 12.0.1 how the language is changed? or is dependent to HOS language?

**Comment by AmonRaNet** [comment]
> For me 12_X_overlays.zip works fine on Atmosphère 0.19.2 and Firmware 12.0.1 how the language is changed? or is dependent to HOS language?

HOS settings. For testing you can simply rename accordingly needed json file (if you dont want to change HOS settings). You think in-depended custom settings are needed?

**Comment by AmonRaNet** [comment]
> [lng_zh-CN.zip](https://github.com/XorTroll/emuiibo/files/7318858/lng_zh-CN.zip) @AmonRaNet hello, this is the Chinese translation for this PR. Comparing to the former translation, I use language code `zh-CN` instead of `cn` as I read your comment above.

I have applied translation file. Thanks once more!
It is difficult to test with system languadge :) when you dont know what you are selecting. I think "zh-CN" not exist in modern versions and instead only "zh-Hans"	 and "zh-Hant". I have renamed to "zh-Hant" - please correct me if I am wrong. Generally it works, but some characters are not showing properly (I have tried select "zh-Hans" - same result).

a1011418-98fe-4ebc-b645-c53e9eb6c3ba
5ae0062d-61d1-4319-b67e-9ada18af95f9




--


<br class="Apple-interchange-newline">

**Comment by amazingfate** [comment]
@AmonRaNet zh-Hant is Tranditional Chinese, my translation is the Simplified Chinese version, so the file should be renamed to `lng_zh-Hans.json`. I am new to atmosphere developing, would you mind telling me how to build the overlay file so I can test it by myself?

**Comment by amazingfate** [comment]
@AmonRaNet I have built an .ovl by myself. As for the fonts issue, I have resolved by updating to the latest libtesla. [This commit](https://github.com/WerWolv/libtesla/commit/35fe198c6d34189db9c37abf452ab26d0f652911) should be related.

**Comment by XorTroll** [comment]
Is the PR in a proper state to be merged? I'd like to merge it and start adapting some stuff, since that'll be faster than reviewing it

**Comment by ca1e** [comment]
@XorTroll  Hi, I'd like to have access to maintain this repo like close useless issues or merge pr like this if you dont mind. 
And @AmonRaNet could you plz update the libtesla submodule to the latest commit to fix the font issue?
I would happy to contribute and maintain this repo to keep it up to date.

**Comment by ca1e** [comment]
I just built with the latest libtesla and try with ams1.2.5-13.2.0 and it works fine!

**Comment by XorTroll** [comment]
Alright, built it myself and it looks great, I'm merging it and changing some bits of the code to match my/emuiibo's codestyle
Thanks for all the work, the overlay looks amazing ;)

**Pull request merged**